extensions/eclipse-che-theia-remote-impl-k8s/src/node/k8s-plugin-service-impl.ts

```diff
@@ -0,0 +1,739 @@
+    if (!component.attributes) {
+      component.attributes = {};
+    }
+
+    // ensure component -> attributes -> app.kubernetes.io/name: mysql
+    if (!component.attributes['app.kubernetes.io/name']) {
+      component.attributes['app.kubernetes.io/name'] = component.name;
+    }
+
+    // ensure component -> attributes -> che-theia.eclipse.org/vscode-extensions
+    if (!component.attributes['che-theia.eclipse.org/vscode-extensions']) {
+      component.attributes['che-theia.eclipse.org/vscode-extensions'] = [];
+    }
+
+    // ensure component -> attributes -> che-theia.eclipse.org/vscode-preferences
+    if (!component.attributes['che-theia.eclipse.org/vscode-preferences']) {
+      component.attributes['che-theia.eclipse.org/vscode-preferences'] = {};
+    }
+
+    // ensure component -> container -> args
+    if (!component.container.args) {
+      component.container.args = ['sh', '-c', '${PLUGIN_REMOTE_ENDPOINT_EXECUTABLE}'];
+    }
+
+    // ensure component -> container -> env
+    if (!component.container.env) {
+      component.container.env = [];
+    }
+
+    // ensure component -> container -> env :: { name: 'PLUGIN_REMOTE_ENDPOINT_EXECUTABLE', value: '...' }
+    const pluginRemoteEndpointExecutable = component.container.env.find(
+      value => value.name === 'PLUGIN_REMOTE_ENDPOINT_EXECUTABLE'
+    );
+    if (!pluginRemoteEndpointExecutable) {
+      component.container.env.push({
+        name: 'PLUGIN_REMOTE_ENDPOINT_EXECUTABLE',
+        value: '/remote-endpoint/plugin-remote-endpoint',
+      });
+    }
+
+    // ensure component -> container -> env :: { name: 'THEIA_PLUGINS', value: '...' }
+    const theiaPlugins = component.container.env.find(value => value.name === 'THEIA_PLUGINS');
+    if (!theiaPlugins) {
+      component.container.env.push({
+        name: 'THEIA_PLUGINS',
+        value: `local-dir:///plugins/sidecars/${component.name}`,
+      });
+    }
+
+    // ensure component -> container -> sourceMapping
+    if (!component.container.sourceMapping) {
+      component.container.sourceMapping = '/projects';
+    }
+
+    // ensure component -> container -> volumeMounts
+    if (!component.container.volumeMounts) {
+      component.container.volumeMounts = [];
+    }
+
+    // ensure component -> container -> volumeMounts :: { name: 'remote-endpoint', path: '/remote-endpoint' }
+    const remoteEndpoint = component.container.volumeMounts.find(value => value.name === 'remote-endpoint');
+    if (!remoteEndpoint) {
+      component.container.volumeMounts.push({
+        name: 'remote-endpoint',
+        path: '/remote-endpoint',
+      });
+    }
+
+    // ensure component -> container -> volumeMounts :: { name: 'plugins', path: '/plugins' }
+    const plugins = component.container.volumeMounts.find(value => value.name === 'plugins');
+    if (!plugins) {
+      component.container.volumeMounts.push({
+        name: 'plugins',
+        path: '/plugins',
+      });
+    }
+  }
```

Comment:
This method should not do all this stuff.
it looks like it's doing what Dashboard is adding

Reply:
Done